check_api/pom.xml

```diff
@@ -181,6 +181,22 @@
               </jdkToolchain>
             </configuration>
           </execution>
+          <execution>
+            <id>java23</id>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+            <configuration>
+              <jdkToolchain>
+                <version>23</version>
+              </jdkToolchain>
+              <compileSourceRoots>
+                <compileSourceRoot>${basedir}/src/main/java23</compileSourceRoot>
+              </compileSourceRoots>
+              <!-- multiReleaseOutput requires setting release -->
+              <outputDirectory>${project.build.outputDirectory}/META-INF/versions/23</outputDirectory>
+            </configuration>
+          </execution>
```

Comment:
A few things are of note here:
- Without `<goal>compile</goal>` this execution is skipped. This also means that the `java24` execution below is currently a no-op.
- This configuration doesn't override the inherited  `<source>17</source>` and `<target>17</target>` settings, though it could. (If so, we should do the same below.)
- I'm not sure each of these executions should use an exactly-matching toolchain JDK version. Perhaps it's better to use the latest JDK whenever possible, so as to minimize the number of JDKs required.